File: app/src/main/java/app/akilesh/nex/MainActivity.java
package app.akilesh.nex;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;


import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.PackageManager;
import android.content.pm.ResolveInfo;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;


import com.google.android.material.bottomnavigation.BottomNavigationView;


import java.util.List;

import static android.view.View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR;
import static android.view.View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
import static android.view.WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS;



public class MainActivity extends AppCompatActivity implements HomeFragment.OnFragmentInteractionListener, AboutFragment.OnFragmentInteractionListener, HelpFragment.OnFragmentInteractionListener, ThemeFragment.OnFragmentInteractionListener {
    int modeType;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(this);
        int themeID = sharedPref.getInt("ThemePrefs",1);
        AppCompatDelegate.setDefaultNightMode(themeID);


        View decorView = getWindow().getDecorView();
        decorView.setSystemUiVisibility(FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
        modeType=AppCompatDelegate.getDefaultNightMode();
        if(modeType == AppCompatDelegate.MODE_NIGHT_NO) {
            decorView.setSystemUiVisibility(SYSTEM_UI_FLAG_LIGHT_STATUS_BAR | SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR);

        }

        //
        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);
        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                Fragment fragment = null;
                switch (item.getItemId()) {
                    case R.id.menu_home:
                        fragment = HomeFragment.newInstance();
                        break;
                    case R.id.menu_info:
                        fragment = AboutFragment.newInstance();
                        break;
                    case R.id.menu_help:
                        fragment = HelpFragment.newInstance();
                        break;

                    case R.id.menu_invert:
                        fragment = ThemeFragment.newInstance();
                        break;


                }
                if (fragment != null) {
                    FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
                    fragmentTransaction.replace(R.id.frameLayout, fragment);
                    fragmentTransaction.commit();
                }
                return true;
            }
        });

        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.frameLayout, HomeFragment.newInstance());
        fragmentTransaction.commit();


    }


    private boolean isCallable(Intent intent) {
        List<ResolveInfo> list = getPackageManager().queryIntentActivities(intent,
                PackageManager.MATCH_DEFAULT_ONLY);
        return list.size() > 0;
    }

    public void launchJunkCleaner(View view) {
        Intent intent = new Intent(Intent.ACTION_MAIN);
        intent.setClassName("com.evenwell.cleaner", "com.evenwell.cleaner.MainActivity");
        if (isCallable(intent)) startActivity(intent);
        else Toast.makeText(this, "Junk Cleaner is not installed", Toast.LENGTH_SHORT).show();

    }


    public void launchSmartBoost(View view) {
        Intent intent = new Intent(Intent.ACTION_MAIN);
        intent.setClassName("com.evenwell.smartboost", "com.evenwell.smartboost.activities.MainActivity");
        if (isCallable(intent)) startActivity(intent);
        else Toast.makeText(this, "Smart Boost is not installed", Toast.LENGTH_SHORT).show();

    }

    public void launchTaskManager(View view) {
        Intent intent = new Intent(Intent.ACTION_MAIN);
        intent.setClassName("com.evenwell.memorycleaner", "com.evenwell.memorycleaner.MainActivity");
        if (isCallable(intent)) startActivity(intent);
        else Toast.makeText(this, "Task Manager is not installed", Toast.LENGTH_SHORT).show();

    }

    public void launchVirusScan(View view) {
        Intent intent = new Intent(Intent.ACTION_MAIN);
        intent.setClassName("com.evenwell.viruscan", "com.evenwell.viruscan.qscanner.VirusScannerActivity");
        if (isCallable(intent)) startActivity(intent);
        else Toast.makeText(this, "Virus Scanner is not installed", Toast.LENGTH_SHORT).show();

    }

    public void launchAIFloatingTouch(View view) {
        Intent intent = new Intent(Intent.ACTION_MAIN);
        intent.setClassName("com.nbc.AIFloatingTouch", "com.nbc.AIFloatingTouch.STouchActivity");
        if (isCallable(intent)) startActivity(intent);
        else Toast.makeText(this, "AI Floating Touch is not installed", Toast.LENGTH_SHORT).show();

    }

    public void launchScreenRecorder(View view) {
        Intent intent = new Intent(Intent.ACTION_MAIN);
        intent.setClassName("com.nbc.android.screenrecord", "com.nbc.android.screenrecord.ui.ActivitySetting");
        if (isCallable(intent)) startActivity(intent);
        else Toast.makeText(this, "AI Floating Touch is not installed", Toast.LENGTH_SHORT).show();

    }

    public void launchFireWall(View view) {
        Intent intent = new Intent(Intent.ACTION_MAIN);
        intent.setClassName("com.evenwell.firewall", "com.evenwell.firewall.TrafficControl");
        if (isCallable(intent)) startActivity(intent);
        else Toast.makeText(this, "App Traffic Control is not installed", Toast.LENGTH_SHORT).show();

    }

}

File: app/src/main/java/app/akilesh/nex/MainActivity.kt
package app.akilesh.nex


import android.content.res.Configuration
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
import android.view.View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
import android.view.WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import app.akilesh.nex.fragments.BottomNavigationDrawerFragment
import app.akilesh.nex.fragments.HomeFragment
import app.akilesh.nex.fragments.SettingsFragment
import com.google.android.material.bottomappbar.BottomAppBar
import com.google.android.material.floatingactionbutton.FloatingActionButton


class MainActivity : AppCompatActivity() {

    private var currentNightMode: Int = -1
    private val brand: String = Build.BRAND
    private lateinit var bottomAppBar: BottomAppBar


    override fun onCreate(savedInstanceState: Bundle?) {

        if(brand != "Nokia") {
            Toast.makeText(this, "This app is only for Nokia phones!", Toast.LENGTH_LONG).show()
            this.finishAffinity()
        }

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        bottomAppBar = findViewById(R.id.bar)
        setSupportActionBar(bottomAppBar)

        val decorView = window.decorView
        decorView.systemUiVisibility = FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS
        currentNightMode = resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
        when (currentNightMode) {
            Configuration.UI_MODE_NIGHT_NO -> {
                val colorAccent = ContextCompat.getColor(this, R.color.colorAccent)
                window.navigationBarColor = colorAccent
                window.statusBarColor = colorAccent
                decorView.systemUiVisibility = SYSTEM_UI_FLAG_LIGHT_STATUS_BAR or SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
            }

            Configuration.UI_MODE_NIGHT_YES -> {
                val colorSurface = Color.parseColor("#121212")
                window.statusBarColor = colorSurface
                window.navigationBarColor = colorSurface
            }

        }

        if(savedInstanceState == null)
            showFragment(HomeFragment.TAG)

        val fab = findViewById<FloatingActionButton>(R.id.fab)
        fab.setOnClickListener {
            showFragment(HomeFragment.TAG)
        }

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.bottom_appbar_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when(item?.itemId){
            android.R.id.home -> {
                val bottomNavDrawerFragment = BottomNavigationDrawerFragment()
                bottomNavDrawerFragment.show(supportFragmentManager, bottomNavDrawerFragment.tag)
            }

            R.id.app_bar_settings -> {
                showFragment(SettingsFragment.TAG)
            }
        }
        return true
    }

    private fun showFragment(tag: String) {
        var fragment = supportFragmentManager.findFragmentByTag(tag)
        if (fragment == null) {
            when (tag) {
                HomeFragment.TAG -> {
                    fragment = HomeFragment()
                }
                SettingsFragment.TAG -> {
                    fragment = SettingsFragment()
                }
            }
        }

        fragment?.let {
            supportFragmentManager
                .beginTransaction()
                .replace(R.id.frameLayout, it, tag)
                .commit()
        }
    }

}


